### PR TITLE
fix: forceful update of server state variable in handleClient changes for multiple reset changes scenario

### PR DESCRIPTION
**Pull request by AmirthavalliG**
During the second time or any number of times when the reset changes (edit scenario of same resource) is invoked , oDataServerState is not reinitialized properly causing inconsistencies  (actual server state data is lost)
resetChanges

### CODE REVIEW

File: src/sap/fhir/model/r4/FHIRModel.js [hunk 1]
```diff
@@ -953,6 +953,10 @@ sap.ui.define([
 		}
 		if (!vServerValue && oRequestInfo.method === HTTPMethod.PUT) {
 			this._setProperty(this.oDataServerState, aResPath, FHIRUtils.deepClone(oResource), true);
+		} else if (vServerValue && oRequestInfo.method === HTTPMethod.PUT && deepEqual(vServerValue, oResource)) {
+			// special handling when the server data and the client data before applying the change is the same(after multiple reset changes)
```
Comment (flovogt) [inline src/sap/fhir/model/r4/FHIRModel.js:957]: add a whitespace before bracket. `is the same (after`

File: src/sap/fhir/model/r4/FHIRModel.js [hunk 2]
```diff
@@ -953,6 +953,10 @@ sap.ui.define([
 		}
 		if (!vServerValue && oRequestInfo.method === HTTPMethod.PUT) {
```
Comment (flovogt) [inline src/sap/fhir/model/r4/FHIRModel.js:954]: We could create a new private function `_isServerDateUpToDate` where we check if an update is needed

``
FHIRModel.prototype._isServerDateUpToDate = function(vServerValue, oResource, sHTTPMethod){
  if (!vServerValue && sHTTPMethod=== HTTPMethod.PUT) {
    return false;
  } else if (vServerValue &&  sHTTPMethod === HTTPMethod.PUT && deepEqual(vServerValue, oResource)) {
    return false;
  }
  return true;
}
``

and in the `_handleClientChanges ` method we have then:
``
if(!this._isServerDateUpToDate(vServerValue, oResource, oRequestInfo.method)){
  this._setProperty(this.oDataServerState, aResPath, FHIRUtils.deepClone(oResource), true);
}
``

**Review by flovogt — CHANGES_REQUESTED** [review]

File: src/sap/fhir/model/r4/FHIRModel.js (hunk 1 above)
Comment (AmirthavalliG) [reply src/sap/fhir/model/r4/FHIRModel.js]: done!

File: src/sap/fhir/model/r4/FHIRModel.js (hunk 2 above)
Comment (AmirthavalliG) [reply src/sap/fhir/model/r4/FHIRModel.js]: done!

File: src/sap/fhir/model/r4/FHIRModel.js [hunk 3]
```diff
@@ -1602,5 +1602,26 @@ sap.ui.define([
 		return sStrucDefUrl;
 	};
 
+	/**
+	 * Determines whether server state variable needs an update
+	 *
+	 * @param {object} vServerValue Existing server state value
+	 * @param {object} oResource The FHIR resource
+	 * @param {string} sHTTPMethod HTTP Method for the resource
```
Comment (flovogt) [inline src/sap/fhir/model/r4/FHIRModel.js:1610]: can we use here the enum as type

**Review by flovogt — CHANGES_REQUESTED** [review]

File: src/sap/fhir/model/r4/FHIRModel.js (hunk 3 above)
Comment (AmirthavalliG) [reply src/sap/fhir/model/r4/FHIRModel.js]: done

File: src/sap/fhir/model/r4/FHIRModel.js [hunk 4]
```diff
@@ -951,7 +951,7 @@ sap.ui.define([
 		if (sGroupId) {
 			this._setProperty(this.mResourceGroupId, FHIRUtils.deepClone(aResPath), sGroupId, true);
 		}
-		if (!vServerValue && oRequestInfo.method === HTTPMethod.PUT) {
+		if (!this._isServerDateUpToDate(vServerValue, oResource, oRequestInfo.method)) {
```
Comment (flovogt) [inline src/sap/fhir/model/r4/FHIRModel.js:954]: Method name should be `isServerStateUpToDate` isn't`?

**Review by flovogt — CHANGES_REQUESTED** [review]

File: src/sap/fhir/model/r4/FHIRModel.js (hunk 4 above)
Comment (AmirthavalliG) [reply src/sap/fhir/model/r4/FHIRModel.js]: my bad,.. thanks!

Comment (AmirthavalliG) [reply src/sap/fhir/model/r4/FHIRModel.js]: done

**Review by flovogt — APPROVED** [review]
LGTM

**Pull request merged**